doc/system_requirements.md

```diff
@@ -0,0 +1,206 @@
+# System Requirement Specification Advanced Analytics Framework
+
+## Introduction
+
+This document describes the system requirements for the advanced analytics 
+framework on which complex algorithms can be built.
+
+
+## About This Document
+
+### Goal
+
+The goal of this library is to provide a general framework to implement complex 
+data analysis algorithms with Exasol. This framework provides certain features 
+to users in which they are able to run their implementations.
+
+## Stakeholders
+
+Software developers use this framework to develop complex data processing 
+algorithms such as machine learning methods.
+
+## Features
+
+Features are the highest level requirements in this document that describe the 
+main functionality of Advanced Analytics Framework.
+
+### Global Loop
+`feat~global-loop~1`
+
+Some complex algorithms need a global loop. In each iteration of the loop, they 
+need to execute SQL queries until the algorithm no longer needs to run 
+further queries. 
+
+Needs: req
+
+### Implementation Framework
+`feat~implementation-framework~1`
+
+The logic of the algorithm is implemented by user. The implementation needs 
+to call a framework which provides certain features to the user code, such as 
+automatic state handling. 
+
+Needs: req
+
+
+### Error Handling
+`feat~error-handling~1`
+
+Any errors, that may occur during state transitions or  the execution of queries, 
+should be handled. Temporarily created files and tables need to be cleaned.
+User can set option to keep temporary records. It can be useful to keep these 
+for debugging.
+
+Needs: req
+
+
+## Functional Requirements
+
+This section lists functional requirements. The requirements are grouped by 
+feature where they belong to a single feature.
+
+
+
+### Global Loop
+
+#### Initiating Global Loop
+`req~intiating-global-loop~1`
+
+It expects the configuration of the operation, which is implemented by user, as 
+input. It needs to build starting query using this configuration. 
+
+Covers:
+
+* [feat~global-loop~1](#global-loop)
+
+Needs: dsn
+
+#### Iterating over Loop
+`req~iterating-over-loop~1`
+
+At each iteration, the framework checks the output of the called operation query 
+to decide whether to continue. If the output contains the necessary SQLs and the 
+status information indicating that it is in progress, these SQLs should run as 
+the next state action and perform the new state transition. 
```

Comment:
```suggestion
to decide whether to continue. If the user code decides to stop the framework returns the results returned by the user code. Otherwise, the user code can start another iteration with further queries.
```